pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +375,463 @@ def Sphere(
+        A monotonically increasing sequence of values specifying radial
+        points. Must have at least two points.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying theta
+        points in degrees. Must have at least two points.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying phi
+        points in degrees. Must have at least two points.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees phi.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.SolidSphere: Sphere creation using linear sampling.
+    pyvista.Sphere: Sphere that describes outer 2D surface.
+
+    Examples
+    --------
+    Create a solid sphere using default parameters.  The
+    default creates a sphere that is linearly sampled with
+    respect to spherical coordinates, the same as
+    :func:`pyvista.SolidSphere`.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> solid_sphere = pv.SolidSphereGeneric()
+    >>> solid_sphere.plot(show_edges=True)
+
+    Linearly sampling spherical coordinates does not lead to
+    cells of all the same size at each radial position.  In
+    particular, cells near the poles have smaller sizes.
+    Plot the cell volumes of a solid sphere with only one layer of
+    cells.
+
+    >>> solid_sphere = pv.SolidSphereGeneric(
+    ...     radius=np.linspace(0, 0.5, 2),
+    ...     theta=np.linspace(180, 360, 30),
+    ...     phi=np.linspace(0, 180, 30),
+    ... )
+    >>> solid_sphere = solid_sphere.compute_cell_sizes()
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    Sampling the polar angle in a nonlinear manner allows for consistent cell volumes.  See
+    `Sphere Point Picking <https://mathworld.wolfram.com/SpherePointPicking.html>`_.
+
+    >>> phi = np.rad2deg(np.arccos(np.linspace(1, -1, 30)))
+    >>> solid_sphere = pv.SolidSphereGeneric(
+    ...     radius=np.linspace(0, 0.5, 2),
+    ...     theta=np.linspace(180, 360, 30),
+    ...     phi=phi,
+    ... )
+    >>> solid_sphere = solid_sphere.compute_cell_sizes()
+    >>> solid_sphere.plot(
+    ...     scalars="Volume", show_edges=True, clim=[3e-5, 5e-4]
+    ... )
+
+    """
+    radius = np.asanyarray(radius)
+    theta = np.asanyarray(theta)
+    phi = np.asanyarray(phi)
+
+    nr = len(radius)
+    ntheta = len(theta)
+    nphi = len(phi)
+
+    if nr < 2:
+        raise ValueError("radius resolution must be 2 or more")
+    if ntheta < 2:
+        raise ValueError("theta resolution must be 2 or more")
+    if nphi < 2:
+        raise ValueError("phi resolution must be 2 or more")
```

Comment:
It is easier to read if the error and conditional expressions are the same.
```suggestion
    if not nr >= 2:
        raise ValueError("radius resolution must be 2 or more")
    if not ntheta >= 2:
        raise ValueError("theta resolution must be 2 or more")
    if not nphi >= 2:
        raise ValueError("phi resolution must be 2 or more")
```

Reply:
Not so much after the post, it seems to me that it is not an important difference. As-is will be fine.

Reply:
Yeah, no, please don't do this. Adding an unnecessary negation to a conditional makes it a lot harder to argue about.